### PR TITLE
Preserve memory layout when aten batchnorm is used

### PR DESCRIPTION
**Pull request by eqy**
#62594 

CC @cpuhrsch

### CODE REVIEW

**Comment by cpuhrsch** [comment]
Thanks! Patching now to verify. On a side note, looks very similar to https://github.com/pytorch/pytorch/pull/62482.

**Review by cpuhrsch — APPROVED** [review]
Resolves the issue.

File: test/test_nn.py [hunk 1]
```diff
@@ -9067,6 +9067,17 @@ def test_batchnorm_buffer_update_when_stats_are_not_tracked(self):
         self.assertTrue(torch.equal(running_mean, bn.running_mean))
         self.assertTrue(torch.equal(running_var, bn.running_var))
 
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA not available")
+    def test_batchnorm_nhwc_cuda(self):
+        (N, C, H, W) = 2, 64, 50, 50
+        model = torch.nn.BatchNorm2d(C, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+        model = model.eval().cuda().half()
+        inp1 = torch.randn(N, C, H, W, device=torch.device('cuda'), dtype=torch.half)
```
Comment (cpuhrsch) [inline test/test_nn.py:9075]: While we're at it, do we want to span a wider net for dtypes, i.e. float32? Or is that already covered?

Comment (eqy) [reply test/test_nn.py:9075]: That might depend on the intent of the test. If I understand correctly, float32 with channels last would dispatch to cudnn batchnorm rather than aten batchnorm.

Comment (cpuhrsch) [reply test/test_nn.py:9075]: I'd say it could be worth it to generally cover channels last support for BatchNorm2d. If float32 isn't already covered by another test, then extending this one could prevent future regressions / errors. I'd assume that the fallback kernel should work whether cudnn is available or not, so it's not needed to guard against cudnn being available for this test.

File: aten/src/ATen/native/cuda/Normalization.cu [hunk 2]
```diff
@@ -422,7 +422,7 @@ std::tuple<Tensor&, Tensor&, Tensor&> batch_norm_cuda_out(const Tensor& self, co
 }
 
 std::tuple<Tensor, Tensor, Tensor> batch_norm_cuda(const Tensor& self, const c10::optional<Tensor>& weight_opt, const c10::optional<Tensor>& bias_opt, const c10::optional<Tensor>& running_mean_opt, const c10::optional<Tensor>& running_var_opt, bool train, double momentum, double epsilon) {
-  auto output = at::empty_like(self, at::MemoryFormat::Contiguous);
+  auto output = at::empty_like(self, self.suggest_memory_format());
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/Normalization.cu:425]: can you also add a check in `batch_norm_elementwise` channels-last case that `output` is channels-last contiguous?

Comment (cpuhrsch) [reply aten/src/ATen/native/cuda/Normalization.cu:425]: cc @eqy

Comment (eqy) [reply aten/src/ATen/native/cuda/Normalization.cu:425]: followup in #62839
